### PR TITLE
Check if Binary Tree is Symmetric in C++

### PR DESCRIPTION
**Pull request by Ashee-Kanungo**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

I have added the code in c++ to check if given binary tree is symmetric or not.

### Checklist

- yes I've read the contribution guidelines.
- yes I've checked the issue list before deciding what to submit.
- yes I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#2753

### CODE REVIEW

**Comment by Ashee-Kanungo** [comment]
@ankitaggarwal23 @gat786 @anushkrishnav @siddharth25pandey @HarshCasper please review this PR.

File: C-Plus-Plus/cp/Binary_Tree_Symmetric_or_not.cpp [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+TreeNode* newNode(int n){
+	TreeNode* t = new TreeNode;
+	t->value = n;
+	t->left = NULL;
+	t->right = NULL;
+	return t;
+}
+
+int height(TreeNode* A){		//this function calculate the height of the Binary Tree
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/cp/Binary_Tree_Symmetric_or_not.cpp]: Shift the comment above the function

File: C-Plus-Plus/cp/Binary_Tree_Symmetric_or_not.cpp [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+}
+
+int height(TreeNode* A){		//this function calculate the height of the Binary Tree   
+	if(A==NULL){
+		return 0;
+	}
+	return 1 + max(height(A->left),height(A->right));
+}
+
+void level_order(TreeNode* A, int h, vector<int> &v){ 		//This gives us the vector of values of all the nodes at the given height
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/cp/Binary_Tree_Symmetric_or_not.cpp]: Shift the comment above the function

File: C-Plus-Plus/cp/Binary_Tree_Symmetric_or_not.cpp [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+	}
+	if(h==1){
+		v.push_back(A->value);
+		return;
+	}
+	level_order(A->left,h-1,v);
+	level_order(A->right,h-1,v);
+}
+
+bool is_level_palindrome(TreeNode* A, int h){		// This function checks if all the levels of given binary tree forms palindrome
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/cp/Binary_Tree_Symmetric_or_not.cpp]: Shift the comment above the function

File: C-Plus-Plus/cp/Binary_Tree_Symmetric_or_not.cpp [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+	level_order(A,h,v);
+	for(int i=0;i<v.size()-1;i++){
+		if(v[i]!=v[v.size()-1-i]){
+			return false;
+		}
+	}
+	return true;
+}
+
+bool Check_Symmetry(TreeNode* A){		//This function checks if given binary tree is symmetrical
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/cp/Binary_Tree_Symmetric_or_not.cpp]: Shift the comment above the function

File: C-Plus-Plus/cp/Binary_Tree_Symmetric_or_not.cpp [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+This binary tree is symmetric, so output should be - "Given binary tree is symmetric."
+
+Suppose if input binary tree is:
+    10
+   / \
+  6   6
+ /   / 
+4    4 
+This binary tree is not symmetric, so output will be - "Given binary tree is not symmetric."
+*/
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/cp/Binary_Tree_Symmetric_or_not.cpp]: - Remove the extra testcase one is enough. 
- Add Time and Space Complexity at the end of the Code   
- Add new line at the EOF

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

**Comment by Ashee-Kanungo** [comment]
@ankitaggarwal23 I have made all the changes in the PR. Please review it.

File: C-Plus-Plus/cp/Binary_Tree_Symmetric_or_not.cpp [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+
+Here we have given following binary tree as input:
+    5
+   / \
+  6   6
+ / \ / \
+3  8 8  3 
+This binary tree is symmetric, so output should be - "Given binary tree is symmetric."
+*/
+
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/cp/Binary_Tree_Symmetric_or_not.cpp:103]: Remove extra one line at the end

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

**Comment by Ashee-Kanungo** [comment]
@ankitaggarwal23 I have removed one extra line at the end.

**Comment by Ashee-Kanungo** [comment]
@ankitaggarwal23 please review it now.

**Review by ankitaggarwal23 — APPROVED** [review]

**Pull request merged**